sdks/python/apache_beam/transforms/external.py

```diff
@@ -289,6 +311,96 @@ def _has_constructor(self):
         self._constructor_param_kwargs)
 
 
+class SchemaTransformsConfig(object):
+  """
+  Information regarding a SchemaTransform available in an external SDK.
+  """
+  def __init__(self, identifier, schema, named_inputs, named_outputs):
+    self._identifier = identifier
+    self._configuration_schema = schema
+    self._named_inputs = named_inputs
+    self._named_outputs = named_outputs
+
+  @property
+  def identifier(self):
+    return self._identifier
+
+  @property
+  def configuration_schema(self):
+    return self._configuration_schema
+
+  @property
+  def named_inputs(self):
+    return self._named_inputs
+
+  @property
+  def named_outputs(self):
+    return self._named_outputs
+
+
+class SchemaAwareExternalTransform(ptransform.PTransform):
+  """A proxy transform for SchemaTransforms implemented in external SDKs.
+
+  This allows Python pipelines to directly use existing SchemaTransforms
+  available to the expansion service without adding additional code in external
+  SDKs.
+
+  :param identifier: unique identifier of the SchemaTransform.
+  :param expansion_service: (Optional) an expansion service to use.  If none is
+      provided, a default expansion service will be started.
+  :param classpath: (Optional) A list paths to additional jars to place on the
+      expansion service classpath.
+  :kwargs: field name to value mapping for configuring the schema transform.
+      keys map to the field names of the schema of the SchemaTransform
+      (in-order).
+  """
+  def __init__(
+      self, identifier, expansion_service=None, classpath=None, **kwargs):
+    self._expansion_service = expansion_service
+    self._payload_builder = SchemaTransformPayloadBuilder(identifier, **kwargs)
+    self._classpath = classpath
+
+  def expand(self, pcolls):
+    # Register transform with the expansion service and the identifier.
+    # Expand the transform using the expansion service and the config_row.
+    if self._expansion_service is None:
+      self._expansion_service = BeamJarExpansionService(
+          ':sdks:java:expansion-service:app:shadowJar',
+          extra_args=['{{PORT}}'],
+          classpath=self._classpath)
+    return pcolls | ExternalTransform(
+        common_urns.schematransform_based_expand.urn,
+        self._payload_builder,
+        self._expansion_service)
+
+  @staticmethod
+  def discover(expansion_service, regex=None):
```

Comment:
For simplicity I think we can skip the regex filtering until we know we need it.